[Luxon] Improve DX by adding docs

CarsonF opened the pull request:
I've copied all of the JSDocs from the Luxon src to here (excluding types). The aim is to provide a better developer experience when working with Luxon.

I love Luxon and have been using it for years. Yet due to the nature of working with dates and all its complexities, I regularly have questions. i.e. What `toLocaleString` format does what I'm looking for.
Furthermore, Luxon has a lot of terse names for its methods/properties. I'm often unsure the exact purpose of a function, or not trusting of my memory.

Luxon has great docs in its src but my IDE doesn't even bother looking there since there are type definitions. So I either have to manually go find the src method in node_modules/luxon/src/... to view its jsdocs or pull up the website...again.

Having these extra descriptions, examples, links, and key functionality notes helps smooth the experience out & confirm that I'm using the library correctly without having to look externally.

Luxon is a stable library. Its surface API has barely changed in the past 2 years, thus I don't think that copying all this here is  risky or hard to maintain.

Hopefully this was a good use of my afternoon and can collectively save us all time over the next months & years.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

peterblazejewicz commented:
when doing this restructe, can you consider to create `src/luxon.d.tx`:
```ts
export const VERSION: string;

export * from "./datetime";
export * from "./duration";
export * from "./info";
export * from "./interval";
export * from "./settings";
export * from "./zone";
export * from "./misc";
```
and in your `index.d.ts`:
```ts
export * from "./src/luxon";
```
the reason: there is no dependencies in package based on `./src/`, only relative paths.

peterblazejewicz commented on types/luxon/src/info.d.ts:
```diff
@@ -0,0 +1,174 @@
+    numberingSystem?: NumberingSystem;
+}
+
+// @deprecated
```
```suggestion
// deprecated
```
JSDoc won't work, let's be consistent now

CarsonF replied:
Ah If I change it to `/** @deprecated */` then my IDE will show interface usage as deprecated (strikethrough).

CarsonF replied:
Changed the comment type so the tag is recognized

peterblazejewicz commented on types/luxon/src/misc.ts:
```diff
@@ -0,0 +1,63 @@
+    keepCalendarTime?: boolean;
+}
+
+// @deprecated
```
```suggestion
// deprecated
```

peterblazejewicz reviewed (approved):
LGMT!
@CarsonF thx!

CarsonF commented:
Thanks for reviewing @peterblazejewicz! Just double checking usage in my codebase currently.

Not sure if there's a difference but I missed the `.d.ts` on `misc`.

Also wondering if I should try to declare classes as default, as that is how the src is declared.

CarsonF commented:
Ok I renamed the misc file. I tried default classes, but I didn't like it so I reverted. I updated the comment types so the deprecated tag is recognized.

peterblazejewicz commented:
> Also wondering if I should try to declare classes as default, as that is how the src is declared.

Please, don't, focus on single topic.

CarsonF commented:
Ready to merge

peterblazejewicz reviewed (approved):
LGTM!

CarsonF commented:
I just added a few test lines to make the bot happy, though they were legitimate.

CarsonF commented:
Ready to merge
